sklearn/metrics/_classification.py

```diff
@@ -2013,63 +2038,110 @@ class after being classified as negative. This is the case when the
     >>> class_likelihood_ratios(y_true, y_pred, labels=["non-cat", "cat"])
     (1.5, 0.75)
     """
-
+    # TODO(1.8): When `raise_warning` is removed, the default return value with
+    # zero_division="warn" does not need to be "np.nan" anymore and should be the lowest
+    # score for each metric respectively (1 for LR+ and 0 for LR-) to match the other
+    # functions that take a zero_division param. Return values and warning messages need
```

Comment:
After looking at this anew, I see that you meant that the change of the default value for `zero_division` from `warn` to the lowest possible scores needs to be a new deprecation cycle.

Can we also consider doing two steps in one? Because the goal is that the functionality of `zero_division` matches the other functions that take a `zero_division` param and the longer this is not the case, the more confusing.